File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        hSet={}
        maxF=0
        left=0
        res=0
        for right in range(len(s)):
            hSet[s[right]]=1 + hSet.get(s[right],0)
            maxF=max(maxF,hSet[s[right]])
            if right-left + 1 - maxF<=k:
                res=max(res,right-left+1)
            else:
                hSet[s[left]]-=1
                left+=1
        return res

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res= new ArrayList<>();
        Deque<TreeNode> q = new ArrayDeque<>();
        if(root!=null){
            q.add(root);
        }
        else {
            return res;
        }
        while (!q.isEmpty()){
            int size=q.size();
            List<Integer> temp = new ArrayList<>();
            for(int i=0;i<size;i++){
                TreeNode node=q.poll();
                if(node!=null){
                    temp.add(node.val);
                    if(node.left!=null)q.add(node.left);
                    if(node.right!=null)q.add(node.right);
                }
            }
            res.add(temp);
        }
        return res;

    }
}
